src/libsyntax/ptr.rs

```diff
@@ -103,6 +103,12 @@ impl<T: ?Sized> Deref for P<T> {
     }
 }
 
+impl<T: ?Sized> DerefMut for P<T> {
+    fn deref_mut(&mut self) -> &mut T {
+        &mut self.ptr
+    }
+}
```

Comment:
This was intentionally not implemented, but I'm considering ways in which we can make ASTs fully immutable (and interned, I suppose).